edge/pkg/metamanager/metaserver/kubernetes/storage/sqlite/imitator/imitator.go

```diff
@@ -286,6 +291,58 @@ func (s *imitator) Event(msg *model.Message) []watch.Event {
 	return ret
 }
 
+func (s *imitator) Patch(ctx context.Context, pi metaserver.PatchInfo) (runtime.Object, error) {
+	gvr := schema.GroupVersionResource{
+		Group:    pi.Group,
+		Version:  pi.Version,
+		Resource: pi.Resource,
+	}
+	results, err := v2.RawMetaByGVRNN(gvr, pi.Namespace, pi.Name)
+	if err != nil {
+		return nil, fmt.Errorf("failed to query:%s/%s resource from meta_v2, err: %+v", pi.Namespace, pi.Name, err)
+	}
+	if len(*results) == 0 {
+		return nil, fmt.Errorf("not found resource:%s/%s from meta_v2", pi.Namespace, pi.Name)
+	}
+
+	originalResource := new(unstructured.Unstructured)
+	err = json.Unmarshal([]byte((*results)[0].Value), originalResource)
+	defaultScheme := scheme.Scheme
+	GV := schema.GroupVersion{Group: pi.Group, Version: pi.Version}
+	GroupVersionKind := GV.WithKind(originalResource.GetKind())
+	schemaReferenceObj, err := defaultScheme.New(GroupVersionKind)
+	if err != nil {
+		return nil, fmt.Errorf("failed to build schema reference object, err: %+v", err)
+	}
+
+	copyOriginalRes := originalResource.DeepCopyObject()
+	defaulter := runtime.ObjectDefaulter(defaultScheme)
+	updatedResource := new(unstructured.Unstructured)
+	switch pi.PatchType {
+	case types.StrategicMergePatchType:
+		err = patchutil.StrategicPatchObject(ctx, defaulter, originalResource,
+			pi.Data, updatedResource, schemaReferenceObj, "")
+		if err != nil {
+			return nil, fmt.Errorf("failed to patch resource:%s,%s/%s", GroupVersionKind.Kind, pi.Namespace, pi.Name)
+		}
+	default:
+		return nil, fmt.Errorf("not supoort patchType:%s", pi.PatchType)
+	}
+
+	if !reflect.DeepEqual(copyOriginalRes, updatedResource) {
+		err = s.InsertOrUpdateObj(ctx, updatedResource)
+		if err != nil {
+			return nil, fmt.Errorf("failed to update resource:%s,%s/%s into meta_v2 db", GroupVersionKind.Kind, pi.Namespace, pi.Name)
+		}
```

Comment:
```suggestion
		if s.InsertOrUpdateObj(ctx, updatedResource) != nil {
			return nil, fmt.Errorf("failed to update resource:%s,%s/%s into meta_v2 db", GroupVersionKind.Kind, pi.Namespace, pi.Name)
		}
```
Is it appropriate to write like this